cloudconfig/containerinit/container_userdata_test.go

```diff
@@ -349,3 +415,215 @@ func assertUserData(c *gc.C, cloudConf cloudinit.CloudConfig, expected string) {
+	s.runENIScriptWithAllPythons(c, cmd, simpleENIFile, simpleENIFileExpected, 0, 1)
+}
+
+func (s *UserDataSuite) TestENIScriptUnknownMAC(c *gc.C) {
+	cmd := s.createMockCommand(c, []string{unknownMACENIFileIPOutput})
+	s.runENIScriptWithAllPythons(c, cmd, unknownMACENIFile, unknownMACENIFileExpected, 0, 1)
+}
+
+func (s *UserDataSuite) TestENIScriptHotplugFail(c *gc.C) {
+	cmd := s.createMockCommand(c, []string{hotplugENIFileIPOutputPre})
+	s.runENIScriptWithAllPythons(c, cmd, hotplugENIFile, hotplugENIFileExpectedFail, 0, 3)
+}
+
+func (s *UserDataSuite) TestENIScriptHotplugTooLate(c *gc.C) {
+	for _, python := range s.pythonVersions {
+		c.Logf("test using %s", python)
+		ipCommand := s.createMockCommand(c, []string{hotplugENIFileIPOutputPre, hotplugENIFileIPOutputPre, hotplugENIFileIPOutputPre, hotplugENIFileIPOutputPost})
+		s.runENIScript(c, python, ipCommand, hotplugENIFile, hotplugENIFileExpectedFail, 0, 3)
+	}
+}
+
+func (s *UserDataSuite) TestENIScriptHotplug(c *gc.C) {
+	for _, python := range s.pythonVersions {
+		c.Logf("test using %s", python)
+		ipCommand := s.createMockCommand(c, []string{hotplugENIFileIPOutputPre, hotplugENIFileIPOutputPre, hotplugENIFileIPOutputPost})
+		s.runENIScript(c, python, ipCommand, hotplugENIFile, hotplugENIFileExpected, 0, 3)
+	}
+}
+
+func (s *UserDataSuite) runENIScriptWithAllPythons(c *gc.C, ipCommand, input, expectedOutput string, wait, retries int) {
+	for _, python := range s.pythonVersions {
+		c.Logf("test using %s", python)
+		s.runENIScript(c, python, ipCommand, input, expectedOutput, wait, retries)
+	}
+
+}
+
+func (s *UserDataSuite) runENIScript(c *gc.C, pythonBinary, ipCommand, input, expectedOutput string, wait, retries int) {
+	dataFile := filepath.Join(s.tempFolder, "interfaces")
+	scriptFile := filepath.Join(s.tempFolder, "script.py")
+
+	err := ioutil.WriteFile(dataFile, []byte(input), 0644)
+	c.Assert(err, jc.ErrorIsNil, gc.Commentf("Can't write interfaces file"))
+
+	err = ioutil.WriteFile(scriptFile, []byte(containerinit.PopulateNetworkInterfacesScript()), 0755)
+	c.Assert(err, jc.ErrorIsNil, gc.Commentf("Can't write script file"))
+
+	script := fmt.Sprintf("%q %q --interfaces_file %q --command %q --wait %d --retries %d", pythonBinary, scriptFile, dataFile, ipCommand, wait, retries)
+	result, err := exec.RunCommands(exec.RunParams{Commands: script})
+	c.Assert(err, jc.ErrorIsNil, gc.Commentf("script failed unexpectedly - %s", result))
+	data, err := ioutil.ReadFile(dataFile)
+	c.Assert(err, jc.ErrorIsNil, gc.Commentf("can't open parsed interfaces file"))
+	output := string(data)
+	c.Assert(output, gc.Equals, expectedOutput)
+}
+
+func (s *UserDataSuite) createMockCommand(c *gc.C, outputs []string) string {
+	randBytes := make([]byte, 16)
+	rand.Read(randBytes)
+	baseName := hex.EncodeToString(randBytes)
+	basePath := filepath.Join(s.tempFolder, fmt.Sprintf("%s.%d", baseName, 0))
+	script := fmt.Sprintf("#!/bin/bash\ncat %s\n", basePath)
```

Comment:
given all the things about bash and python we should make sure these tests are skipped on Windows. You should be able to find the standard 
if os == 'windows':
 c.Skip(...)

I don't remember the exact syntax offhand.
We should be able to put that in SetupTest/SetupSuite.